src/workspaces/workspaces.service.ts

```diff
@@ -33,7 +36,22 @@ export class WorkspacesService {
   }
 
   async getWorkspace(id: Workspace['id']) {
-    return this.workspaceRepository.findOne({ id });
+    const workspace = await this.workspaceRepository.findOne({ id });
+    if (!workspace) {
+      return new NotFoundException();
+    }
+    const filterOptions = { workspaceId: id };
+    const sortOptions = null;
+    const paginationOptions = { page: 1, limit: 3 };
+    const threeMembers = await this.usersService.findManyWithPagination({
+      filterOptions,
+      sortOptions,
+      paginationOptions,
+    });
+    const workspaceMemberscount = await this.usersService.getCount({
+      workspaceId: id,
+    });
+    return { workspace, threeMembers, workspaceMemberscount };
   }
```

Comment:
Adding this logic at the service layer may result inconstancy or code duplication code within `getWorkspaces`, that's why I think it's better to achieve this at the repository level.
Let's postpone this change for the current pull request and address it in a separate one.